.github/workflows/generate-benchmark-pages.yml

```
@@ -0,0 +1,41 @@
+name: Generate Benchmark Pages
+on:
+  push:
+    branches:
+      - main  #any push to main branch triggers this workflow
+  workflow_dispatch:
+  # Sets permissions of the GITHUB_TOKEN to allow deployment to GitHub Pages
+permissions:
+  contents: write
```

```suggestion
  contents: read
```
I believe this is all we need since removing your steps that pushed to `main`.